### DISCUSSION TITLE
Exporting metrics: is it possible to use otel exporter?

### DISCUSSION THREAD

**Discussion by compscidr** [open-telemetry/opentelemetry-cpp#952] [Q&A] [answered]
I realize that metrics is still alpha / in development. From what I can tell, it doesn't look like it is possible currently to use the otel exporter to export metrics. Is this correct? Is the only current functioning metric exporter the ostream exporter?

It looks like the otel exporter has a different type of `Export`: https://github.com/open-telemetry/opentelemetry-cpp/blob/main/exporters/otlp/include/opentelemetry/exporters/otlp/otlp_grpc_exporter.h#L92

vs the interface needed for metric exporting: https://github.com/open-telemetry/opentelemetry-cpp/blob/main/sdk/include/opentelemetry/sdk/metrics/exporter.h#L30

What is the suggested way to go about implementing this? Modify the existing `OtlpGrpcExporter` with something like multiple inheritance, or create a new `OtlpGrpcMetricExporter` class or something? Is anyone already working on something like this?

Alternatively, do you think it would be easier to just work with the osteam and use something other than std::cout to get the metrics exported? (ideally I'd like to send it into an otel collector, or failing that directly into prometheus). 

Apologies if some of these questions are dumb, I'm still wrapping my head a bit around how it all fits together.

**Comment by lalitb** [comment, marked as answer]
The development for metrics is paused for the reasons mentioned [here](https://github.com/open-telemetry/opentelemetry-cpp#project-status). The current metrics api/sdk implementation is not compliant with the metrics otel specs which are still evolving. This will need a rewrite once the metrics specs are stable. The interface sdk::metrics::exporter.h mentioned in the question will also change in this process.

So, it is not recommended to use current metrics implementation or adding support for otlp exporter as of now.

**Reply by compscidr** [reply]
Got it, thanks very much for the reply!
